src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,47 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+    private LocalDate getLocalDate(String date) {
+        return LocalDate.parse(date, formatter);
+    }
+
+    private boolean isValidDate(LocalDate startDate, LocalDate endDate, String checkedDate) {
+        LocalDate localDate = getLocalDate(checkedDate);
+        return localDate.isAfter(startDate.minusDays(1))
+            && localDate.isBefore(endDate.plusDays(1));
+    }
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        SalaryInfo salaryInfo = new SalaryInfo();
+        LocalDate startLocalDate = salaryInfo.getLocalDate(dateFrom);
+        LocalDate endLocalDate = salaryInfo.getLocalDate(dateTo);
```

Comment:
There is still a critical issue with the usage of SalaryInfo instance. The instance of this class will be already created by tests and then the method getSalaryInfo() will be called on that instance. From this method, you can reach getLocalDate() and isValidDate() directly, without the need to create another instance of SalaryInfo